desktop/src/views/Workspaces/WorkspaceCard.tsx

```diff
@@ -475,89 +378,154 @@ function WorkspaceCardHeader({
-          <Ripple {...rippleProps} color="yellow.500" />
-          <Text {...sharedTextProps} color="yellow.500">
-            Loading
-          </Text>
-        </>
-      )
+    const actionID = workspace.start({ id, ideConfig: { name: ide } })
+    if (!isIDEFixed) {
+      await client.ides.useIDE(ide)
     }
+    navigateToAction(actionID)
+  }
+  const isOpenDisabled = workspace.data?.status === "Busy"
+  const isOpenDisabledReason =
+    "Cannot open this workspace because it is busy. If this doesn't change, try to force delete and recreate it."
+  const [isStartWithHovering, startWithRef] = useHover()
+  const [isPopoverHovering, popoverContentRef] = useHover()
 
-    if (status === "Running") {
-      return (
-        <>
-          <Box {...sharedProps} backgroundColor="green.200" borderColor="green.400" />
-          <Text {...sharedTextProps} color="green.400">
-            Running
-          </Text>
-        </>
-      )
-    }
+  return (
+    <HStack spacing="2" width="full" justifyContent="end">
+      <ButtonGroup isAttached variant="solid-outline">
+        <Tooltip label={isOpenDisabled ? isOpenDisabledReason : undefined}>
+          <Button
+            aria-label="Start workspace"
+            leftIcon={<Icon as={HiOutlineCode} boxSize={5} />}
+            isDisabled={isOpenDisabled}
+            onClick={() => {
+              const actionID = workspace.start({
+                id,
+                ideConfig: { name: ideName ?? ideName ?? null },
+              })
+              navigateToAction(actionID)
+            }}
+            isLoading={isLoading}>
+            Open
+          </Button>
+        </Tooltip>
+        <Menu placement="top">
+          <MenuButton
+            as={IconButton}
+            aria-label="More actions"
+            colorScheme="gray"
+            icon={<Ellipsis transform={"rotate(90deg)"} boxSize={5} />}
+          />
+          <Portal>
+            <MenuList>
+              <Popover
+                isOpen={isStartWithHovering || isPopoverHovering}
+                placement="right"
+                offset={[100, 0]}>
+                <PopoverTrigger>
+                  <MenuItem
+                    ref={startWithRef}
+                    icon={<Play boxSize={4} />}
+                    isDisabled={isOpenDisabled || isLoading}>
+                    <HStack width="full" justifyContent="space-between">
+                      <Text>Start with</Text>
+                      <ChevronRightIcon boxSize={4} />
+                    </HStack>
+                  </MenuItem>
+                </PopoverTrigger>
+                <PopoverContent zIndex="popover" width="fit-content" ref={popoverContentRef}>
+                  {ides?.map((ide) => (
+                    <MenuItem
+                      isDisabled={isOpenDisabled || isLoading}
+                      onClick={handleOpenWithIDEClicked(id, ide.name)}
+                      key={ide.name}
+                      value={ide.name!}
```

Comment:
Can an ide trully not have a name ? What happens when two don't have a name in this case as the handle click triggers a rerender ?

Reply:
should result in a noop for both but generally Ide's _should_ have a name (since we control them atm)